### PR TITLE
write json serialization optimization

### PR DESCRIPTION
**Pull request by moderakh**
the purpose of this PR is to improve json serialization in document write path:

* Reduces multiple rounds of json serialization/deserialization to and from Document
* If partition-key is provided in the request options, avoids deserializing the item to Document.

numbers for inserting 4000_000 documents against a collection with 1M throughput on Ubuntu with 16CPU cores.

### Scenario 1:
**Total time required for inserting 4_000_000 documents each with 1000 fields**

| |  pk passed as request option | pk extracted from document (require json deserialization)  |
|---|---|---|
|total time to insert 4M docs before this PR| 195 seconds | 186 seconds  |
| total time to insert 4M docs after this PR|   123 seconds|  130 seconds|

benchmark invocation cmd:
```bash
java -Xmx8g -Xms8g -Dcosmos.directModeProtocol=Tcp -Dazure.cosmos.directModeProtocol=Tcp -jar /home/moderakh/baseline.jar -serviceEndpoint https://XXX.documents.azure.com:443/ -masterKey YYY -databaseId testdb -collectionId testcol -consistencyLevel Eventual -concurrency 2000 -numberOfOperations 4000000 -operation WriteThroughput -connectionMode Direct -numberOfPreCreatedDocuments 2000 -documentDataFieldSize 1 -documentDataFieldCount 1000
```

### Scenario2:
**Total time required for inserting 4_000_000 documents each with 10 fields**

| |  pk passed as request option | pk extracted from document (require json deserialization)  |
|---|---|---|
|total time to insert 4M docs before this PR| 85 seconds | 85 seconds  |
| total time to insert 4M docs after this PR|   81 seconds|  85 seconds|

benchmark invocation cmd:
```bash
java -Xmx8g -Xms8g -Dcosmos.directModeProtocol=Tcp -Dazure.cosmos.directModeProtocol=Tcp -jar /home/moderakh/baseline.jar -serviceEndpoint https://XXX.documents.azure.com:443/ -masterKey YYY -databaseId testdb -collectionId testcol -consistencyLevel Eventual -concurrency 2000 -numberOfOperations 4000000 -operation WriteThroughput -connectionMode Direct -numberOfPreCreatedDocuments 2000 -documentDataFieldSize 1 -documentDataFieldCount 10
```

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosItemProperties.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosItemProperties.java]: unused imports, please remove them.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java [hunk 2]
```diff
@@ -52,6 +52,10 @@ public static Document documentFromObject(Object document, ObjectMapper mapper)
         return Document.FromObject(document, mapper);
     }
 
+    public static String toJsonString(Object document, ObjectMapper mapper) {
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java]: Just for my understanding: Why is it modeled as string, isn't stream the right abstraction?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java]: where good question.

There are three options:
1. String
2. Byte[]
3. InputStream/OutputStream

Currently everywhere in SDK, String is passed to DocumentServiceRequest. The public constructors of DocumentServiceRequest only accept String. But DSR internally converts String to Byte[] and holds it as Byte array.

I think there should be a potential room for optimization by passing data as Byte[] to DSR, I am planning to look at this after this PR is merged (as this is not specific to document write).

the other option is InputStream and OutputStream. as mentioned above DSR holds data as byte[] array and support for InputStream/OutputStream might not currently be used anywhere or even may not be in working state. The other challenge is converting from InputStream/OuputStream to Flux which might require some work. I don't know yet where we are in that regard.

My plan is to study the perfs for these options after this PR and make the requires changes.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosItemProperties.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosItemProperties.java]: done

**Pull request merged**